js/button.js

```diff
@@ -1,96 +1,93 @@
 
-    setOptions(options) {
-        const defaultOptions = {
-            width: 160,
-            height: 40,
-            texture: null,
-            x: 0,
-            y: 0,
-            fill: 0xFF3300,
-            label: 'Default button',
-            labelStyle: {
-                fill: '#ffffff',
-                fontSize: '16px',
-                fontFamily: 'Arial, sans-serif',
-            },
-            disabled: false,
-        };
-
-        return {...defaultOptions, ...options};
+    _init() {
+        this.buttonMode = true;
+        this._isActive = false;
+        this._isHover = false;
+        this._switchEnabledState(true);
+        this._setEvents();
+        if (this._options.label) {
+            this._addLabel();
+        }
     }
 
-    init() {
-        this.id = this._generateId();
-        this.element = null;
-        this.buttonMode = true;
-        this.interactive = true;
-        this.x = this.options.x;
-        this.y = this.options.y;
-        this.disabled = this.options.disabled;
+    _setEvents() {
+        this.on("pointerover", this._onHover)
+            .on("pointerout", this._onOut)
+            .on("pointerdown", this._onDown)
+            .on("pointerup", this._onUp)
+            .on("pointerupoutside", this._onUp);
+    }
 
-        if(this.options.texture) {
-            this.drawTexture(this.options.texture);
+    _onHover() {
+        this._isHover = true;
+        if (this._isActive) {
             return;
         }
-        this.drawGraphics();
+        this._updateTexture("hover");
     }
 
-    drawTexture(texture) {
-        this.element = new PIXI.Sprite(texture);
-        this.element.width = this.options.width;
-        this.element.height = this.options.height;
-        this.addChild(this.element);
+    _onOut() {
+        this._isHover = false;
+        if (this._isActive) {
+            return;
+        }
+        this._updateTexture("default");
     }
 
-    drawGraphics() {
-        this.element = new PIXI.Graphics();
-        this.element.x = this.x;
-        this.element.y = this.y;
-        this.element.beginFill(this.options.fill);
-        this.element.drawRect(0, 0, this.options.width, this.options.height);
-        this.element.endFill();
+    _onDown() {
+        this._isActive = true;
+        this._updateTexture("down");
+    }
 
-        if(this.options.label) {
-            this.addLabel();
+    _onUp() {
+        this._isActive = false;
+        if (this._isHover) {
+            this._updateTexture("hover");
+            return;
         }
-        this.addChild(this.element);
+        this._updateTexture("default");
     }
 
-    addLabel() {
-        this.label = new PIXI.Text(this.options.label, this.options.labelStyle);
-        this.label.anchor.set(0.5);
-        this.label.x = this.element.width / 2;
-        this.label.y = this.element.height / 2;
-        this.element.addChild(this.label);
+    _switchEnabledState(isEnabled) {
+        if (!isEnabled) {
```

Comment:
Starting conditional statements from negation tend to add unnecessary complication. Consider:
`if(NOT condition) -> option1; else option2`
can always be replaced with `if(condition) -> option2; else option1`
The second is easier to comprehend almost in all possible cases.
This can be simplified even further with ternary operator:
`this._updateTexture(isEnabled ? "default" : "disabled");`
However, in this specific case, consider setting texture and switching flags directly in disable/enable methods. The _switchEnabledState seems to be redundant.